rl4co/envs/routing/cvrp/generator.py

```diff
@@ -0,0 +1,131 @@
+from typing import Union, Callable
+
+import torch
+
+from torch.distributions import Uniform
+from tensordict.tensordict import TensorDict
+
+from rl4co.utils.pylogger import get_pylogger
+from rl4co.envs.common.utils import get_sampler, Generator
+
+log = get_pylogger(__name__)
+
+
+# From Kool et al. 2019, Hottung et al. 2022, Kim et al. 2023
+CAPACITIES = {
+    10: 20.0,
+    15: 25.0,
+    20: 30.0,
+    30: 33.0,
+    40: 37.0,
+    50: 40.0,
+    60: 43.0,
+    75: 45.0,
+    100: 50.0,
+    125: 55.0,
+    150: 60.0,
+    200: 70.0,
+    500: 100.0,
+    1000: 150.0,
+}
+
+
+class CVRPGenerator(Generator):
+    """Data generator for the Capacitated Vehicle Routing Problem (CVRP).
+    Args:
+        num_loc: number of locations (cities) in the VRP, without the depot. (e.g. 10 means 10 locs + 1 depot)
+        min_loc: minimum value for the location coordinates
+        max_loc: maximum value for the location coordinates
+        loc_distribution: distribution for the location coordinates
+        min_demand: minimum value for the demand of each customer
+        max_demand: maximum value for the demand of each customer
+        demand_distribution: distribution for the demand of each customer
+        capacity: capacity of the vehicle
+
+    Returns:
+        A TensorDict with the following keys:
+            locs [batch_size, num_loc, 2]: locations of each customer
+            depot [batch_size, 2]: location of the depot
+            demand [batch_size, num_loc]: demand of each customer
+            capacity [batch_size]: capacity of the vehicle
+    """
+    def __init__(
+        self,
+        num_loc: int = 20,
+        min_loc: float = 0.0,
+        max_loc: float = 1.0,
+        loc_distribution: Union[
+            int, float, str, type, Callable
+        ] = Uniform,
+        depot_distribution: Union[
+            int, float, str, type, Callable
+        ] = Uniform,
+        min_demand: int = 1,
+        max_demand: int = 10,
+        demand_distribution: Union[
+            int, float, type, Callable
+        ] = Uniform,
+        vehicle_capacity: float = 1.0,
```

Comment:
I think `vehicle_capacity` and `capacity` should be the same parameter, is it right?